index.html

```diff
@@ -6372,6 +6379,9 @@ <h2>Changes since JSON-LD Community Group Final Report</h2>
       a context. When this is set, vocabulary-relative IRIs, such as the
       <a>entries</a> of <a>node objects</a>, are expanded or compacted relative
       to the <a>base IRI</a> and the <a>vocabulary mapping</a> using string concatenation.</li>
+    <li>In the <a>LoadDocumentCallback</a>, if the retrieved content is not `application/ld+json`
+      and there is a link header with `rel=alternate` and `type=application/ld+json`, redirect
+      to that content.</li>
```

Comment:
This probably needs to be discussed (it is not just editorial).

This would create a different behaviour when the document is accessed directly, than when it is accessed through the "link header indirection". So I am not a big fan of the idea.

I would rather encourage the author of the JSON-LD document to explicitly set the base to the URL o the non-JSON-LD document, if that is the behaviour they intend.

Reply:
The PR is definitely prospective, as the issue is still marked to be discussed, but I wanted to get it out there and see the consequences. The link header shouldn’t affect any json format, but would affect html content. What document base should be applied can be debated, but using the original document has precedent.

Reply:
Hmm, I think I disagree with @pchampin ... and that we want to treat this like we do with redirects, which I believe uses the original document. The argument for doing what @pchampin suggests would be that there's no way to know ahead of time what the final document will be, so you couldn't set the base to it, whereas you could set the base to the original document if you wanted it to stick.

But, processors could have a flag to switch to this other behavior. It seems to me that the sensible default would be the original document. Especially given the way that we expect this feature to be used ... which is that the linked document is really a hidden implementation detail. If you wanted to use the linked document directly you would have specified it instead of some, e.g., HTML document.

Reply:
👎 to introducing another flag, though. If the document should have some specific base different from the originally referenced resource, it can use `@base`.

My thought in using the original resource base, rather than the referenced document, is essentially the same reasoning that lead to the semantics of HTTP status 303 for range-14 considerations.

Reply:
If this use of `alternate` (which comes from [HTML primarily](https://html.spec.whatwg.org/multipage/links.html#rel-alternate)) is meant to supplant using content negotiation, then the expectation would be (afaict) that the original resource URL would (semantically) be the correct base URL--which I think agrees with what @gkellogg said above.

Once the `alternate` is retrieved it should experientially be *like* getting a content negotiated response with a `Content-Location` set to the `alternate` and having `Location` still be set to the original Request-URI--which is how content negotiation plays out.

Reply:
I agree that the intention is similar to content negotiation, and this advocates for keeping the base IRI of the original document.

However, it is not, as @dlongley puts it, a *hidden* implementation detail (as conneg would be) because the client is given the URL of the actual JSON-LD content. In practice, it is much more similar to a 303 redirect. And I am not aware of any precedent where, when following a 303 redirect or `rel=alternate`, the client is supposed to keep the base IRI of the source document... Any pointer?

Reply:
This is the same mechanism used, e.g. by DBPedia to distinguish a resource from the page about the resource.  In thier case http://dbpedia.org/resource/Richard_III_of_England redirects to http://dbpedia.org/page/Richard_III_of_England so that statements can be made about the resource, rather than the page which describes the resource. (Fragment identifiers are much nicer).

See [Wikipedia article on HTTPRange-14](https://en.wikipedia.org/wiki/HTTPRange-14) and [Jeni's discussion for the TAG](http://www.jenitennison.com/2012/05/11/using-punning-to-answer-httprange-14.html).

I don't know of anything specific for `rel="alternate"` that relates to this, although @BigBlueHat brought something up on the call. IMO, it is reasonable for a format, such as JSON-LD, to ascribe such semantics.

Reply:
I understand the parallel with HTTPRange-14 . But I see nowhere in those documents an indication that relative IRIs in the target document should be resolved against the source document's IRI...

Reply:
To make my point clearer: https://perso.liris.cnrs.fr/pierre-antoine.champin/2019/redirect/ shows that browsers do not keep the source URL of a 303 redirect to interpret relative URLs in the target document (or at least, mine does not).

Reply:
What I mentioned on the call Friday was that the only definition of `rel="alternate"` (that I'm aware of) is in https://html.spec.whatwg.org/multipage/links.html#rel-alternate

The summary definition is "Gives alternate representations of the current document." but the follow-on expansion on it includes "The meaning of this keyword depends on the values of the other attributes."

The examples in that section include [alternate CSS styles](https://html.spec.whatwg.org/multipage/links.html#rel-alternate:rel-alternate-3) (i.e. `rel="alternate stylesheet"`), Atom/RSS alternate representations (i.e. `rel="alternate" type=""`), and using it for language [content] negotiation (i.e. `rel="alternate" hreflang=""` +/- `type=""`).

So, using `alternate` here to accomplish a pre-flight content negotiation (of sorts) is not without its precedence conceptually. However, none of that directly answers the base URI situation...as in the closest experiential case (language negotiation) the browser would navigate to the new resource based on language preferences--thereby acting more like a `30*` redirect than like content negotiation.

Reply:
Additionally, if we stick with `alternate`, we may need to require other link parameters (or type parameters) to clarify that this is a representation of the current document *as a context* vs. a representation of the current document *expressed in JSON-LD* (i.e. as a data document).

We could use the existing context profile URL (which is used as a `rel` in [Interpreting JSON as JSON-LD](https://w3c.github.io/json-ld-syntax/#interpreting-json-as-json-ld)). The experience would be similar to referencing a data document as the context of another data document.

Reply:
> I understand the parallel with HTTPRange-14 . But I see nowhere in those documents an indication that relative IRIs in the target document should be resolved against the source document's IRI...

There's a discussion in [Cool URIs](https://www.w3.org/TR/cooluris/#r303uri):

> ... Each of the RDF documents would contain statements about the appropriate resource, using the original URI, e.g. http://www.example.com/id/alice, to identify the described resource.

Reply:
Yes, but they do not say that a relative URI could or should be used for that...

Reply:
At this point, I think you need to raise a new issue for discussion, because we seem to have quite different views on this, and discussing in a merged PR isn't a good way to be sure we resolve it.